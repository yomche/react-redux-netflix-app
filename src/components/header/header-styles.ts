import styled from 'styled-components';
import { headerBackground } from './images';

export const StyledHeader = styled.header`
    width: 1200px;
    height: 300px;
    background-image: url(${headerBackground});
    background-color: #808080;
    background-size: cover;
    margin: auto;
`;

export const StyledSorter = styled.section`
    width: 1200px;
    height: 40px;
    background-color: #808080;
    margin-top: 20px;
`;
