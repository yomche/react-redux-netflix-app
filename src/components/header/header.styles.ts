import styled from 'styled-components';

const BACKGROUND_COLOR = '#808080';

export const StyledHeader = styled.header`
  width: 1200px;
  background-image: url('../../../assets/netflix-background.jpg');
  background-color: ${BACKGROUND_COLOR};
  background-size: cover;
  margin: auto;
`;
