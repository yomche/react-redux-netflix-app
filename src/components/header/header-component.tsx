import React, { FC } from 'react';
import { StyledHeader, StyledSorter } from './header-styles';
import { Controls } from '../controls/controls-component';
import { SearchInput } from '../search-input/search-input-component';
import { NavPanel } from '../nav-panel/nav-panel-component';

export const HeaderComponent: FC = () => (
  <StyledHeader>
    <NavPanel />
    <SearchInput />
    <Controls />
    <StyledSorter />
  </StyledHeader>
);
