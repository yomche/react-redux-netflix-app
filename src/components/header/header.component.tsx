import React, { FC } from 'react';
import { StyledHeader } from './header.styles';
import { SearchInput } from '../search-input/search-input.component';
import { NavPanel } from '../nav-panel/nav-panel.component';
import { MovieFullInfo } from '../movie-full-info/movie-full-info.component';
import { TypeOfView } from '../../constants/app.constants';
import { CurrentMovieType } from '../../types';

type HeaderProps = {
  onSetSearchInputValue: (value: string) => void;
  viewType: string;
  onSetMovieListType: () => void;
  currentMovie: CurrentMovieType;
  onSetMovieSearchType: (value: string) => void;
  movieSearchType: string;
};

export const Header: FC<HeaderProps> = ({
  viewType,
  currentMovie,
  onSetSearchInputValue,
  onSetMovieListType,
  onSetMovieSearchType,
  movieSearchType,
}) => (
  <StyledHeader>
    <NavPanel onClickToChange={onSetMovieListType} />
    {viewType === TypeOfView.movieList ? (
      <>
        <SearchInput
          updateSearchInputValue={onSetSearchInputValue}
          setMovieSearchType={onSetMovieSearchType}
          movieSearchType={movieSearchType}
        />
      </>
    ) : (
      <MovieFullInfo currentMovieFullInfo={currentMovie} />
    )}
  </StyledHeader>
);
