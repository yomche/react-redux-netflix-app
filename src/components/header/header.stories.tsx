import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs, optionsKnob as options } from '@storybook/addon-knobs';
import { Header } from './header.component';

storiesOf('Knobs/Header', module)
  .addDecorator(withKnobs)
  .add('different view types', () => {
    const viewType = {
      movieList: 'movieList',
      MovieFullInfo: 'movieFullInfo',
    };
    const viewTypeSelect = options('viewType', viewType, 'movieList', { display: 'select' });
    const currentMovie = {
      posterPath:
        'https://sun9-72.userapi.com/impf/a-V1H8zLHXqzMb3UY8yzgicFlOVZkHw4TNXeLQ/PHLkRY64ehg.jpg?size=1459x2160&quality=96&sign=1063b2565af2a89c8ae442b61427f9e7&type=album',
      title: 'Black Mirror',
      releaseDate: '2011-02-07',
      overview:
        'Set in a world only minutes from our own, Black Mirror unveils how modern technologies can backfire and be used against their makers, every episode set in a slightly different reality with different characters combating different types of technologies.',
      id: 0,
    };
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    function setMovieListType() {}

    return (
      <Header
        viewType={viewTypeSelect}
        currentMovie={currentMovie}
        onSetMovieListType={setMovieListType}
      />
    );
  });
