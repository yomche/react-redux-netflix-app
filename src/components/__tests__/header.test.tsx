import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import { Header } from '../header/header.component';

describe('< Header /> component', () => {
  const HeaderMock = {
    viewType: 'movieList',
    movieSearchType: 'title',
    currentMovie: {
      poster_path: 'Test URL',
      title: 'Test title',
      release_date: 'Test release date',
      overview: 'Test overview',
      id: 0,
    },
    onSetMovieListType: jest.fn(),
    onSetSearchInputValue: jest.fn(),
    setMovieSearchType: jest.fn(),
  };
  test('renders correctly', () => {
    const { asFragment } = render(
      <Router>
        <Header
          viewType={HeaderMock.viewType}
          currentMovie={HeaderMock.currentMovie}
          onSetMovieListType={HeaderMock.onSetMovieListType}
          movieSearchType={HeaderMock.movieSearchType}
          onSetMovieSearchType={HeaderMock.setMovieSearchType}
        />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
