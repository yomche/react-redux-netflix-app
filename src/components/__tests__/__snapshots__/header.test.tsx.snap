// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< Header /> component renders correctly 1`] = `
<DocumentFragment>
  <header
    class="StyledHeader-sc-10yfg31-0 liWvUy"
  >
    <nav
      class="StyledNav-sc-1ix25fi-0 iRDFNW"
    >
      <div
        class="StyledLogo-sc-1wiya5k-0 ehakiy"
      />
      <a
        href="/"
      >
        <button
          class="StyledSearchButton-sc-1ix25fi-1 bxvkPE"
        >
           SEARCH 
        </button>
      </a>
    </nav>
    <form
      class="StyledSearchForm-sc-1uqbsb-0 bMzDOT"
    >
      <h1
        class="StyledHeading-sc-1uqbsb-3 jjwjis"
      >
        FIND YOUR MOVIE
      </h1>
      <div
        class="StyledInput-sc-1uqbsb-1 jrqfiq"
      >
        <input
          name="inputValue"
          placeholder="Quentin Tarantino"
          type="text"
          value=""
        />
      </div>
      <section
        class="StyledControlsSection-sc-1uqbsb-5 llnDDy"
      >
        <label
          class="StyledTitleButton-sc-1uqbsb-6 ipEjeb"
        >
          <input
            checked=""
            name="searchType"
            type="radio"
            value="title"
          />
           TITLE
        </label>
        <label
          class="StyledGenreButton-sc-1uqbsb-7 XQUYS"
        >
          <input
            name="searchType"
            type="radio"
            value="genres"
          />
           GENRE
        </label>
        <button
          class="StyledSubmitButton-sc-1uqbsb-8 kfslJL"
          type="submit"
        >
           SUBMIT 
        </button>
      </section>
    </form>
  </header>
</DocumentFragment>
`;
